_posts/2018-03-23-minnowboard-turbot-remote-firmware-flashing-with-remote-testing-environment.md

```diff
@@ -0,0 +1,312 @@
+author: Arkadiusz Cichocki
+post_excerpt: ""
+layout: post
+published: true
+post_date: 2018-03-23 12:00:00
+tags:
+  - RTE
+  - Minnowboard Turbot
+  - Robot Framework
+  - firmware
+  - ROM
+  - remote development
+  - remote testing
+categories:
+
+---
+
+# Minnowboard Turbot  remote firmware flashing with RTE (Remote Testing Environment)
+
+
+Introduction
+------------
+
+Work related to a hardware carries some restrictions which don't occur when
+working only with a software. One of them is limited number of devices.
+This one may cause a problem with a accessibility to the platform. Limited 
+number of users could slow development and testing.
+What is more work with a harware requires a minimal knowledge about theory 
+of circuits and signals to elimite platform damage by an user. 
+Hardware can be expensive too.
+
+Remote Testing Environment project was made to resolve mentioned problems.
+The result of work is shown below:
+![](https://3mdeb.com/wp-content/uploads/2017/07/rte1-e1521807540813.jpg)
+![](https://3mdeb.com/wp-content/uploads/2017/07/rte2-e1521807599613.jpg)
+
+That platform makes possible to work on a hardware remotely.
+RTE connected to the tested device can provide some more advantages like:
+* Possibility to built testing system only once and place it in separated
+and secure place.
+* Advantage described above allows to keep order and eliminate time wasting
+for items search and connect.
+* Reduced number of hardware requirements for the user. Now he needs only a
+computer with a net connection and installed tools.
+
+Selected RTE functionality:
+* SPI with a header,
+* I2C with a header,
+* RS232,
+* OC buffer outputs header,
+* GPIO,
+* USB,
+* built-in relay with DC Jack 2.5/5.5 mm connectors to DUT power supply control.
+
+RTE + Minnowboard Turbot
+------------------------
+
+To provide remote access to the Minnowboard Turbot platform it was connected to
+the RTE.
+
+### Connection
+
+![](https://3mdeb.com/wp-content/uploads/2017/07/rte-mwb-e1521807566346.jpg)
+
+* Power supply
+
+5 V power supply for Minnowboard Turbot was connected to the RTE `J12` 
+connector. Then DC Jack - DC Jack wire was connected to the `J13` of RTE and
+to the `J9` of Minnowboard. That configuration allows to control power supply
+remotely.
+
+* SPI
+
+| Minnowboard J1 | Color         | RTE                      |
+|:--------------:|:-------------:|:------------------------:|
+| 1              | red           | 1 (J9)                   |
+| 2              | black         | 2 (J1)                   |
+| 3              | orange        | 3 (J1) + 1.2 kΩ resistor |
+| 4              | green         | 4 (J1)                   |
+| 5              | blue          | 5 (J1)                   |
+| 6              | brown         | 6 (J1)                   |
+| 7              | -             | NC                       |
+| 8              | -             | NC                       |
```

Comment:
@acihy I'm not sure if such table will render correctly. @mscibisz ?

Reply:
It should work according to: https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#tables

Reply:
@acihy @mscibisz let's try that. I assume with the new infrastructure it should be easy to test if this works. Let's proceed with merging when all requests addressed.